ipatests/test_integration/test_installation.py

```diff
@@ -1916,3 +1917,73 @@ def test_replica0_install(self):
+
+    @classmethod
+    def install(cls, mh):
+        pass
+
+    def test_install_master_withalgo_sha384withrsa(self, server_cleanup):
+        tasks.install_master(
+            self.master,
+            extra_args=['--ca-signing-algorithm=SHA384withRSA'],
+        )
+
+        # check Signing Algorithm post installation
+        dashed_domain = self.master.domain.realm.replace(".", '-')
+        cmd_args = ['certutil', '-L', '-d',
+                    '/etc/dirsrv/slapd-{}/'.format(dashed_domain),
+                    '-n', 'Server-Cert']
+        result = self.master.run_command(cmd_args)
+        assert 'SHA-384 With RSA Encryption' in result.stdout_text
+
+    def test_install_master_modify_existing(self, server_cleanup):
+        """
+        Setup a master
+        Stop services
+        Modify default.params.signingAlg in CS.cfg
+        Restart services
+        Resubmit certs (Certs should still have earlier Algo)
+        Issue a new Cert
+        """
+        tasks.install_master(self.master)
+        self.master.run_command(['ipactl', 'stop'])
+        cs_cfg_content = self.master.get_file_contents(paths.CA_CS_CFG_PATH,
+                                                       encoding='utf-8')
+        new_lines = []
+        replace_str = "ca.signing.defaultSigningAlgorithm=SHA384withRSA"
+        ocsp_rep_str = "ca.ocsp_signing.defaultSigningAlgorithm=SHA384withRSA"
+        for line in cs_cfg_content.split('\n'):
+            if line.startswith('ca.signing.defaultSigningAlgorithm'):
+                new_lines.append(replace_str)
+            elif line.startswith('ca.ocsp_signing.defaultSigningAlgorithm'):
+                new_lines.append(ocsp_rep_str)
+            else:
+                new_lines.append(line)
+        self.master.put_file_contents(paths.CA_CS_CFG_PATH,
+                                      '\n'.join(new_lines))
+        self.master.run_command(['ipactl', 'start'])
+
+        cmd = ['getcert', 'list', '-f', paths.RA_AGENT_PEM]
+        result = self.master.run_command(cmd)
+        request_id = get_certmonger_fs_id(result.stdout_text)
+
+        # resubmit RA Agent cert
+        cmd = ['getcert', 'resubmit', '-f', paths.RA_AGENT_PEM]
+        self.master.run_command(cmd)
+
+        tasks.wait_for_certmonger_status(self.master,
+                                         ('CA_WORKING', 'MONITORING'),
+                                         request_id)
+
+        self.master.run_command(['pki', '-d', '/tmp/certdb/',
+                                 '-c ', 'Secret123', 'client-init', '--force'])
+        self.master.run_command(['pki', '-d', '/tmp/certdb',
+                                 '-c ', 'Secret123', 'client-cert-import',
+                                 'IPA RA', '--cert', paths.RA_AGENT_PEM])
+        cmd_args = ['certutil', '-d', '/tmp/certdb', '-L', '-n', 'IPA RA']
+        result = self.master.run_command(cmd_args)
+        assert 'SHA-256 With RSA Encryption' in result.stdout_text
```

Comment:
It may be simpler to display the cert using openssl:
`openssl x509 -noout -text -in /var/lib/ipa/ra-agent.pem`

The signature will appear as:
`    Signature Algorithm: sha256WithRSAEncryption`

Reply:
Changed it to openssl